--- origsrc/Python-3.0b3/setup.py	2008-07-17 11:23:53.000000000 -0500
+++ src/Python-3.0b3/setup.py	2008-08-20 22:27:08.000000000 -0500
@@ -398,7 +398,7 @@
 
         # Check for MacOS X, which doesn't need libm.a at all
         math_libs = ['m']
-        if platform in ['darwin', 'mac']:
+        if platform in ['darwin', 'mac', 'cygwin']:
             math_libs = []
 
         # XXX Omitted modules: gl, pure, dl, SGI-specific modules
