--- origsrc/Python-3.1/setup.py	2009-07-22 21:54:05.961037100 -0500
+++ src/Python-3.1/setup.py	2009-07-22 21:55:40.701234700 -0500
@@ -894,7 +894,7 @@ class PyBuildExt(build_ext):
             missing.append('_sqlite3')
 
         # The standard Unix dbm module:
-        if platform not in ['cygwin']:
+        if platform not in ['win32']:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg.split('=')[-1] for arg in config_args
