--- origsrc/Python-3.2.3/Lib/ctypes/__init__.py	2012-04-11 01:54:02.000000000 -0500
+++ src/Python-3.2.3/Lib/ctypes/__init__.py	2012-07-06 01:02:29.711890700 -0500
@@ -436,7 +436,8 @@ pydll = LibraryLoader(PyDLL)
 if _os.name in ("nt", "ce"):
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
 elif _sys.platform == "cygwin":
-    pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+    pythonapi = PyDLL("libpython%d.%d%s.dll" % \
+                      (_sys.version_info[:2] + tuple(_sys.abiflags)))
 else:
     pythonapi = PyDLL(None)
 
