--- origsrc/Python-3.2.5/configure.ac	2013-07-30 16:39:31.743160000 -0500
+++ src/Python-3.2.5/configure.ac	2013-07-30 18:41:19.375132600 -0500
@@ -3773,7 +3773,7 @@ AC_MSG_RESULT($SOABI)
 
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
-    Linux*|GNU*)
+    Linux*|GNU*|CYGWIN*)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
--- origsrc/Python-3.2.3/Python/dynload_shlib.c	2012-04-11 01:54:08.000000000 -0500
+++ src/Python-3.2.3/Python/dynload_shlib.c	2012-07-19 20:28:01.691090700 -0500
@@ -38,6 +38,8 @@
 
 const struct filedescr _PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    {"." SOABI ".dll", "rb", C_EXTENSION},
+    {"module." SOABI ".dll", "rb", C_EXTENSION},
     {".dll", "rb", C_EXTENSION},
     {"module.dll", "rb", C_EXTENSION},
 #else  /* !__CYGWIN__ */
